.github/workflows/scripts/setup-cluster.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -e
+
+source ./scripts/version
+
+if [ -z "$CLUSTER_NAME" ]; then
+    echo "CLUSTER_NAME must be specified when setting up a cluster"
+    exit 1
+fi
+
+if [ -z "$K3S_VERSION" ]; then
+  echo "K3S_VERSION must be specified when setting up a cluster, use `k3d version list k3s` to find valid versions"
+  exit 1
+fi
+
+# waits until all nodes are ready
+wait_for_nodes(){
+  echo "wait until all agents are ready"
+  while :
```

Comment:
My preference is to support a timeout and fail if it's reached. I would go with 5 x an observed time on a non-state-of-the-art system.